components/tools/OmeroPy/src/omero/gateway/__init__.py

```diff
@@ -7111,6 +7112,58 @@ class _ImageWrapper (BlitzObjectWrapper, OmeroRestrictionWrapper):
     PLANEDEF = omero.romio.XY
 
     @classmethod
+    def _getQueryString(cls, opts=None):
+        """
+        Extend base query to handle filtering of Images by Datasets.
+        Returns a tuple of (query, clauses, params).
+        Supported opts: 'dataset': <dataset_id> to filter by Dataset
+                        'load_pixels': <bool> to load Pixel objects.
+                        'load_channels': <bool> to load Channels and
+                                                    Logical Channels
+                        'orphaned': <bool>. Images not in Dataset or WellSample
+
+        :param opts:        Dictionary of optional parameters.
+        :return:            Tuple of string, list, ParametersI
+        """
+        query, clauses, params = super(
+            _ImageWrapper, cls)._getQueryString(opts)
+        if opts is not None and 'dataset' in opts:
+            query += ' join obj.datasetLinks dlink'
+            clauses.append('dlink.parent.id = :did')
+            params.add('did', rlong(opts['dataset']))
+        load_pixels = False
+        load_channels = False
+        orphaned = False
+        if opts is not None:
+            load_pixels = 'load_pixels' in opts and opts['load_pixels']
+            load_channels = 'load_channels' in opts and opts['load_channels']
+            orphaned = 'orphaned' in opts and opts['orphaned']
+        if load_pixels or load_channels:
+            # We use 'left outer join', since we still want images if no pixels
+            query += ' left outer join fetch obj.pixels pixels'
+        if load_channels:
+            query += ' join fetch pixels.channels as channels' \
+                     ' join fetch channels.logicalChannel'
```

Comment:
If we want to know information about the logical channel, it is also necessary to fetch
``photometricInterpretation, illumination, mode and contrastMethod`` from the logical channel